core/controllers/base_test.py

```diff
@@ -1423,3 +1462,195 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
         with self.swap(self, 'testapp', self.mock_testapp):
             self.post_json(
                 '/correctmock', payload, expected_status_int=200)
+
```

Comment:
Please add a set of tests for testing the functionality of the schema validation with a test handler. All the test cases need not be covered but the working of the schema validation has to tested.

Reply:
Done

Reply:
The coverage is still not 100% for this file. Also please add a test for the schema validation of args of GET request, so that all the schema validation is completely tested.

Reply:
Done, ptal